configure.ac

```diff
@@ -646,15 +646,15 @@ AC_CHECK_DECLS([__builtin_clz, __builtin_clzl, __builtin_clzll])
 dnl Check for MSG_NOSIGNAL
 AC_MSG_CHECKING(for MSG_NOSIGNAL)
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <sys/socket.h>]],
- [[ int f = MSG_NOSIGNAL; ]])],
+ [[ return MSG_NOSIGNAL; ]])],
```

Comment:
This seems broken to me in the first place, isn't `MSG_NOSIGNAL` a preprocessor define? If so there's no need to check for it in configure.ac, we can just use `#if defined()` in `net.cpp` and `netbase.cpp`.

Reply:
`MSG_NOSIGNAL` is a macro in my environment and maybe in lots of other environments too. So this check can be wiped away from `configure.ac` and `#ifdef MSG_NOSIGNAL` be used instead of `#ifdef HAVE_MSG_NOSIGNAL` in the source code. On my environment that will work.

I guess the reason for this check in `configure.ac` is that we are not guaranteed that it will be a macro. Maybe somewhere it could be `const int MSG_NOSIGNAL = ...`?

My change intends to fix an "unused variable" warning, with a minimum chance of breaking something else.

Reply:
I believe all of these options are required to be macros by the standard (precisely for reasons like this). Also note that `const int FOO = 1` and `#define FOO 1` are not the same, using a const value is strictly worse as it requires declaring an extern value that gets storage space allocated in the program. So there's no reason an implementation would ever do things that way.

The problem with keeping logic like this "just in case" is that these `AC_COMPILE_IFELSE` checks are relatively expensive, so we should only be using them for features that actually require compilation.

Reply:
You are right - `const int` requires storage somewhere. I have removed the checks from `configure.ac` and updated the pull request.